skimage/util/compare.py

```diff
@@ -44,7 +44,7 @@ def compare_images(image1, image2, method='diff', *, n_tiles=(8, 8)):
     elif method == 'blend':
         comparison = 0.5 * (img2 + img1)
     elif method == 'checkerboard':
-        shapex, shapey = img1.shape
```

Comment:
Thanks this seems useful! Though we should probably add a test for this specific case and add a `channel_axis` parameter here instead of this fix. `channel_axis` is something we've settled on to address these kind of cases. It's more explicit and gives more control to the user. Would you be willing to make these changes? :)

Long-term we could even add ND support to this function. Right now its even undocumented what kind of images this function supports.

Reply:
sure I'd be happy to do both of those things once I'm done with my uni finals this week.